cli/internal/fs/path.go

```diff
@@ -116,38 +113,19 @@ func (ap AbsolutePath) RelativePathString(path string) (string, error) {
 	return filepath.Rel(ap.asString(), path)
 }
 
-// EnsureDirFS ensures that the directory containing the given filename is created
-func EnsureDirFS(fs afero.Fs, filename AbsolutePath) error {
-	dir := filename.Dir()
-	err := fs.MkdirAll(dir.asString(), DirPermissions)
-	if errors.Is(err, syscall.ENOTDIR) {
-		err = fs.Remove(dir.asString())
-		if err != nil {
-			return errors.Wrapf(err, "removing existing file at %v before creating directories", dir)
-		}
-		err = fs.MkdirAll(dir.asString(), DirPermissions)
-		if err != nil {
-			return err
-		}
-	} else if err != nil {
-		return errors.Wrapf(err, "creating directories at %v", dir)
-	}
-	return nil
-}
-
 // WriteFile writes the given bytes to the specified file
-func WriteFile(fs afero.Fs, filename AbsolutePath, toWrite []byte, mode os.FileMode) error {
-	return afero.WriteFile(fs, filename.asString(), toWrite, mode)
+func WriteFile(filename AbsolutePath, toWrite []byte, mode os.FileMode) error {
+	return os.WriteFile(filename.asString(), toWrite, mode)
 }
 
 // ReadFile reads the contents of the specified file
-func ReadFile(fs afero.Fs, filename AbsolutePath) ([]byte, error) {
-	return afero.ReadFile(fs, filename.asString())
+func ReadFile(filename AbsolutePath) ([]byte, error) {
+	return os.ReadFile(filename.asString())
 }
 
 // RemoveFile removes the file at the given path
-func RemoveFile(fs afero.Fs, filename AbsolutePath) error {
-	return fs.Remove(filename.asString())
+func RemoveFile(filename AbsolutePath) error {
```

Comment:
We should implement `AbsolutePath.Remove` and delete this. Also, at least a few places in tests are calling `os.Remove(someAbsPath.ToString())` which both doesn't use this function and also doesn't show up as something to be migrated.

Reply:
I've cleaned up a chunk of them; we currently have 20 calls to `ToString()` (not sure if they're all `AbsolutePath`s) and 21 `ToStringDuringMigration()`. I've left the ones which aren't connected to removing `afero`.